### ISSUE TITLE
SNOW-1014403: ERROR: Failed building wheel for snowflake-connector-python

### ISSUE THREAD

**Issue by najibfahs** [snowflakedb/snowflake-sqlalchemy#462]
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?

  Python 3.12.0 (tags/v3.12.0:0fb18b0, Oct  2 2023, 13:03:39) [MSC v.1935 64 bit (AMD64)]

2. What operating system and processor architecture are you using?

   Windows-11-10.0.22621-SP0

3. What are the component versions in the environment (`pip freeze`)?

   anyio==4.2.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==2.4.1
async-lru==2.0.4
attrs==23.2.0
Babel==2.14.0
beautifulsoup4==4.12.3
bleach==6.1.0
certifi==2023.11.17
cffi==1.16.0
charset-normalizer==3.3.2
colorama==0.4.6
comm==0.2.1
contourpy==1.2.0
cycler==0.12.1
debugpy==1.8.0
decorator==5.1.1
defusedxml==0.7.1
executing==2.0.1
fastjsonschema==2.19.1
fonttools==4.47.2
fqdn==1.5.1
greenlet==3.0.3
idna==3.6
ipykernel==6.29.0
ipython==8.20.0
isoduration==20.11.0
jedi==0.19.1
Jinja2==3.1.3
json5==0.9.14
jsonpointer==2.4
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
jupyter-events==0.9.0
jupyter-lsp==2.2.2
jupyter_client==8.6.0
jupyter_core==5.7.1
jupyter_server==2.12.5
jupyter_server_terminals==0.5.1
jupyterlab==4.0.11
jupyterlab_pygments==0.3.0
jupyterlab_server==2.25.2
kiwisolver==1.4.5
MarkupSafe==2.1.4
matplotlib==3.8.2
matplotlib-inline==0.1.6
mistune==3.0.2
nbclient==0.9.0
nbconvert==7.14.2
nbformat==5.9.2
nest-asyncio==1.5.9
notebook==7.0.7
notebook_shim==0.2.3
numpy==1.26.3
overrides==7.4.0
packaging==23.2
pandas==2.1.4
pandocfilters==1.5.1
parso==0.8.3
pillow==10.2.0
platformdirs==4.1.0
prometheus-client==0.19.0
prompt-toolkit==3.0.43
psutil==5.9.8
pure-eval==0.2.2
pycparser==2.21
Pygments==2.17.2
pyodbc==5.0.1
pyparsing==3.1.1
python-dateutil==2.8.2
python-json-logger==2.0.7
pytz==2023.3.post1
pywin32==306
pywinpty==2.0.12
PyYAML==6.0.1
pyzmq==25.1.2
referencing==0.32.1
requests==2.31.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rpds-py==0.17.1
Send2Trash==1.8.2
setuptools==68.2.2
six==1.16.0
sniffio==1.3.0
soupsieve==2.5
SQLAlchemy==2.0.25
stack-data==0.6.3
terminado==0.18.0
tinycss2==1.2.1
tornado==6.4
traitlets==5.14.1
types-python-dateutil==2.8.19.20240106
typing_extensions==4.9.0
tzdata==2023.4
uri-template==1.3.0
urllib3==2.1.0
wcwidth==0.2.13
webcolors==1.13
webencodings==0.5.1
websocket-client==1.7.0

4. What did you do?

   1 - I installed SQLAlchemy and it was a success
   2- pipenv install --upgrade snowflake-sqlalchemy
then this installation fails:
Installation Succeeded
Pipfile.lock (c43238) out of date, updating to (f03552)...                                                                                                           
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
[=== ] Locking...
.......
ERROR: Failed building wheel for snowflake-connector-python
Failed to build snowflake-connector-python
ERROR: Could not build wheels for snowflake-connector-python, which is required to install pyproject.toml-based projects
[snowflake connector error.txt](https://github.com/snowflakedb/snowflake-sqlalchemy/files/13999790/snowflake.connector.error.txt)

5. What did you expect to see?
successful installation
   What should have happened and what happened instead?

6. Can you set logging to DEBUG and collect the logs?
no
[snowflake connector error.txt](https://github.com/snowflakedb/snowflake-sqlalchemy/files/13999792/snowflake.connector.error.txt)

   ```
   import logging
   import os

   for logger_name in ['snowflake.sqlalchemy', 'snowflake.connector']:
      logger = logging.getLogger(logger_name)
      logger.setLevel(logging.DEBUG)
      ch = logging.StreamHandler()
      ch.setLevel(logging.DEBUG)
      ch.setFormatter(logging.Formatter('%(asctime)s - %(threadName)s %(filename)s:%(lineno)d - %(funcName)s() - %(levelname)s - %(message)s'))
      logger.addHandler(ch)
   ```

**Comment by najibfahs** [comment]
addition info:
I created a new python environment on the same laptop to rule out any conflicts with the pipfiles, but I still get the same error.
I used a different laptop that has Python 3.10.4 and the installation of snowflake-sqlalchemy works (1.5.1) and it installed SQLAchemy 1.4.51 and snowflake-connector-python 3.6.0.

**Comment by sfc-gh-sfan** [comment]
The reason is that `snowflake-connector-python` does not support python 3.12 yet.

The good news is that 3.12 support was already merged, and will be part of 3.7.0 release once it goes out (In progress changelog could be found [here](https://github.com/snowflakedb/snowflake-connector-python/blob/main/DESCRIPTION.md)). Please stay tuned!

**Comment by sfc-gh-sfan** [comment]
Going to close this issue because it is unrelated to `snowflake-sqlalchemy`. Feel free to re-open if you have further questions!

**Comment by najibfahs** [comment]
thank you Shixuan Fan. 
I went to https://pypi.org/project/snowflake-connector-python/ and v3.7.0(January 25,2024) is released with support for Python 3.12:
Added a new boolean parameter force_return_table to SnowflakeCursor.fetch_arrow_all to force returning pyarrow.Table in case of zero rows.
Cleanup some C++ code warnings and performance issues.
Added support for Python 3.12
